packages/components/src/components/CodeEditor/profilerExtension.ts

```diff
@@ -0,0 +1,58 @@
+import { RangeSet, RangeSetBuilder } from "@codemirror/state";
+import { Decoration, EditorView } from "@codemirror/view";
+
+import { simulationField } from "./fields.js";
+
+const makeMark = (intensity: number) => {
+  const min = 100;
+  const color = 255 - min + min * (1 - intensity);
+  return Decoration.mark({
+    attributes: {
+      style: `background-color: rgb(255,${color},${color},50%)`,
+    },
+  });
+};
+
+export function profilerExtension() {
+  const empty = RangeSet.of<Decoration>([]);
+
+  const decorations = EditorView.decorations.compute(
+    ["doc", simulationField.field],
+    (state) => {
+      const simulation = state.field(simulationField.field);
+      if (!simulation) {
+        return empty;
+      }
+
+      if (simulation.code !== state.doc.toString()) {
+        return empty;
+      }
+
+      const { output } = simulation;
+      if (!output.ok) {
+        return empty;
+      }
+
+      const profile = output.value.raw.runOutput.profile;
+      if (!profile) {
+        return empty;
+      }
+
+      const times = profile.times;
+      const totalTimes = times.reduce((a, b) => a + b, 0);
```

Comment:
do you think this is better than some `sum` function?

Reply:
I was in a hurry, otherwise I'd move this to `squiggle-lang`, as a method on `RunProfile` class.

But also tbh I think this is completely fine, assuming both me and you can read it without extra effort, which I think is true. (`lodash/sum` would also be fine)